Mono.Debugging/Mono.Debugging.Client/StackFrame.cs

```diff
@@ -135,7 +135,16 @@ internal void Attach (DebuggerSession debugSession)
 		/// </summary>
 		public string GetLocationSignature ()
 		{
-			return $"{Path.GetFileNameWithoutExtension (this.FullModuleName)}!{this.SourceLocation.MethodName}:{this.Address}";
+			string methodName = this.SourceLocation.MethodName;
+			if (methodName != null && !methodName.Contains("!")) {
+				methodName = $"{Path.GetFileName (this.FullModuleName)}!{methodName}";
+			}
+
+			if (this.Address != 0) {
+				methodName = $"{methodName}:{Address}";
```

Comment:
Should the CurrentLocationSignature be a struct { location, method token }?

Reply:
No, let’s keep it simple and flexible.

Reply:
Define flexible here. Because the problem is:

1. You can't define case sensitivity for a language - i.e. VB
2. You can't define case sensitivity based on filesystem

> APFS: Uses the APFS format. Choose this option if you don’t need an encrypted or case-sensitive format.
> APFS (Encrypted): Uses the APFS format and encrypts the volume.
> APFS (Case-sensitive): Uses the APFS format and is case-sensitive to file and folder names. For example, folders named “Homework” and “HOMEWORK” are two different folders.
> APFS (Case-sensitive, Encrypted): Uses the APFS format, is case-sensitive to file and folder names, and encrypts the volume. For example, folders named “Homework” and “HOMEWORK” are two different folders.